zkvm/docs/zkvm-spec.md

```diff
@@ -1985,13 +1967,8 @@ UTXO IDs are not generally known without fuller view on transaction flow. This c
 
 That said, for contracts created from another contract, the contract ID is determined locally by the parent contract’s ID.
 
+### Do we need qty/flavor introspection ops?
+Previously, we thought we needed them to reblind the received value. However, we normally use `borrow` instead of placing some bounds on a received value. If we only ever mix all values and borrow necessary payments, we do not need to expose these variables at all.
```

Comment:
Correction: reblinding and constraints are orthogonal concepts:

1. "reblinding" was a (now scrapped) protocol to "encrypt a value to a pubkey" in one tx, and then decrypt it in another tx. We will use elgamal encryptions and codify encrypted values or data in specific types, so we won't be operating on individual variables or commitments. Therefore qty/flavor introspection won't be needed there.
2. "constraints on values" are still relevant as always, but with `borrow` we do not need to inspect a Value object — we can simply constrain raw commitments/variables and then `borrow` a value with such commitment. Hence, no need for qty/flavor for this use-case.

Reply:
Thanks, this was super helpful!